compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java

```diff
@@ -311,7 +321,10 @@ private void validateAnnotation(SyntaxNodeAnalysisContext ctx, MetadataNode meta
                     }
                     this.hasAutoIncrementAnnotation = true;
                 }
-                if (annotation.annotReference().toSourceCode().trim().equals(Constants.RELATION)) {
+            } else if (annotation.annotReference().toSourceCode().trim().equals(Constants.RELATION)) {
+                if (mappingConstructorExpressionNode.isPresent()) {
```

Comment:
Is there a possibility this can be false? What is the expected behaviour then?

Reply:
Yes, When we add the annotation without field. 
Eg: 
```@persist:AutoIncrement
    readonly int needId = -1;
```

```
 @persist:Relation
 MedicalItem medicalItem;
````

No need to validate any thing.

Reply:
Ack